File: app/src/main/java/com/rachelleignacio/listbucket/presentation/presenters/AbstractPresenter.java
package com.rachelleignacio.listbucket.presentation.presenters;

import com.rachelleignacio.listbucket.domain.executor.MainThread;
import com.rachelleignacio.listbucket.domain.executor.ThreadExecutor;

/**
 * Created by rachelleignacio on 3/31/17.
 */

public abstract class AbstractPresenter {
    protected ThreadExecutor threadExecutor;
    protected MainThread mainThread;

    public AbstractPresenter(ThreadExecutor threadExecutor, MainThread mainThread) {
        this.threadExecutor = threadExecutor;
        this.mainThread = mainThread;
    }
}

File: app/src/main/kotlin/com/rachelleignacio/listbucket/presentation/presenters/AbstractPresenter.kt
package com.rachelleignacio.listbucket.presentation.presenters

import com.rachelleignacio.listbucket.domain.executor.MainThread
import com.rachelleignacio.listbucket.domain.executor.ThreadExecutor

/**
 * Created by rachelleignacio on 8/23/17.
 */
abstract class AbstractPresenter(protected val threadExecutor: ThreadExecutor,
                                 protected val mainThread: MainThread)